app/Services/Registration/Factories/UserFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services\Registration\Factories;
+
+use App\DTO\CreateUserResource;
+use App\Enums\UserStatusesEnum;
+use App\Models\User;
+use App\Repositories\Interfaces\UserRepositoryInterface;
+use App\Services\Registration\Interfaces\Factories\UserFactoryInterface;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Hash;
+
+final class UserFactory implements UserFactoryInterface
+{
+    public function __construct(private UserRepositoryInterface $userRepository)
```

Comment:
Here as well, for example.

```suggestion
    public function __construct(private UserRepositoryInterface $userRepository)If we are doing a dependency injection we should instantiate under construct, for example. 

```suggestion
        private UserRepositoryInterface $userRepository;

    public function __construct(UserRepositoryInterface $userRepository) {
        $this->userRepository = $userRepository;
    }
```
```